common-ui-compose/src/main/java/app/tivi/common/compose/Flow.kt

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+@file:Suppress("NOTHING_TO_INLINE")
+
+package app.tivi.common.compose
+
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.State
+import androidx.compose.runtime.produceState
+import androidx.compose.ui.platform.LocalLifecycleOwner
+import androidx.lifecycle.Lifecycle
+import androidx.lifecycle.flowWithLifecycle
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.collect
+
+@Composable
+fun <T> Flow<T>.collectAsStateWithLifecycle(
+    initial: T,
+    minActiveState: Lifecycle.State = Lifecycle.State.STARTED,
+): State<T> {
+    val lifecycleOwner = LocalLifecycleOwner.current
+    return produceState(initialValue = initial, this, lifecycleOwner) {
+        flowWithLifecycle(
+            lifecycle = lifecycleOwner.lifecycle,
+            minActiveState = minActiveState
+        ).collect {
+            value = it
+        }
+    }
+}
+
+@Composable
+inline fun <T> Flow<T>.flowWithLocalLifecycle(
```

Comment:
Does this need to be remembered for paging? If so, you might want to call it `inline fun rememberFlowWithLocalLifecycle(flow)

Reply:
It actually works fine right now, but I like the look of `rememberFlowWithLifecycle` so migrated all calls to it